phd-tests/framework/src/test_vm/mod.rs

```diff
@@ -623,14 +623,20 @@ impl TestVm {
     pub fn run_shell_command(&self, cmd: &str) -> Result<String> {
         self.send_serial_str(cmd)?;
 
-        let mut echo_cmd = cmd.to_string();
+        // If the command is multi-line, it won't be echoed literally.
+        // instead, it will (probably) have each line begin with an `>`. so,
```

Comment:
Another in my growing list of reasons to give the `GuestOs` adapters more input into how shell commands actually get issued and get echoed. No need to do that in this PR though.

Reply:
Yup, I also thought about that. For now, I just wanted to unbreak multiline, but we should probably have a nicer way of handling this...